lib/utils/autocomplete.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const homedir = require('os').homedir();
+const path = require('path');
+const md5 = require('md5');
+const readFileSync = require('./fs/readFileSync');
+const fileExistsSync = require('./fs/fileExistsSync');
+
+const Serverless = require('../Serverless');
+
+const tab = require('tabtab')({
+  name: 'serverless',
+});
+
+const getSuggestions = (cacheFilePath) => {
+  const cacheFile = readFileSync(cacheFilePath);
+
+  tab.on('serverless', (data, done) => {
+    done(null, Object.keys(cacheFile.commands));
+  });
+
+  Object.keys(cacheFile).forEach(command => {
+    tab.on(command, (data, done) => {
+      done(null, cacheFile.commands[command]);
+    });
+  });
+
+  tab.start();
+};
+
+const autocomplete = () => {
+  let servicePath = process.cwd();
+  if (!fileExistsSync(path.join(servicePath, 'serverless.yml'))) {
```

Comment:
@pmuens had a good point. I think we should abstract the search process for the `serverless.*` file out to a until function. By doing this we could ensure consistent lookup no matter where it's used. We could also turn cache the lookup results to prevent having to go back to disk on a second call.